EzMap.Domain/Repositories/PoiRepository.cs

```diff
@@ -7,58 +7,61 @@ namespace EzMap.Domain.Repositories;
 
 public interface IPoiRepository
 {
-    void AddPoi(PoiCreateDto dto);
+    void AddPoi(PoiCreateDto dto,CancellationToken token = default);
 
-    Task ReadPoisAsync(CancellationToken token = default);
+    Task<List<Poi>?> GetListPoiAsync(Guid? userId, CancellationToken token = default);
 
-    Task UpdatePoiAsync(PoiUpdateDto dto);
+    Task UpdatePoiAsync(PoiUpdateDto dto,CancellationToken token = default);
 
-    Task DeletePoiAsync(Guid id);
-}
+    Task DeletePoiAsync(Guid id,CancellationToken token = default);
 
+    Task<Poi?> GetPoiById(Guid? userId, Guid id, CancellationToken token = default);
+}
 
 public class PoiRepository : IPoiRepository
 {
-    private readonly EzMapContext _dbContext ;
+    private readonly EzMapContext _dbContext;
 
     public PoiRepository(EzMapContext dbContext)
     {
         _dbContext = dbContext;
     }
 
-    public async Task ReadPoisAsync(CancellationToken token = default)
+    public async Task<Poi?> GetPoiById(Guid? userId ,Guid id, CancellationToken token = default)
     {
-        string query = _dbContext.Pois.ToQueryString();
-        Console.WriteLine();
-        List<Poi> pois = await _dbContext.Pois.ToListAsync(token);
-        foreach (var p in pois)
-        {
-            Console.WriteLine($"{p.Address} {p.Name}");
-        }
-        
-        Console.WriteLine();
+        var poi = await _dbContext.Pois.FirstOrDefaultAsync(x => x.Id == id && x.UserId == userId, cancellationToken: token);
+        return poi;
     }
 
-    public void AddPoi(PoiCreateDto dto)
+    public async Task<List<Poi>?> GetListPoiAsync( Guid? userId ,CancellationToken token = default)
     {
-        Poi poi = new Poi(dto.Name, dto.Address, dto.UserId) ; 
-        _dbContext.Pois.Add(poi);
+        if (userId == null) return null;
+        List<Poi>? listPoi = await _dbContext.Pois.Where(
+            x => x.UserId == userId).ToListAsync(cancellationToken: token);
+
+        return listPoi;
     }
 
-    public async Task UpdatePoiAsync(PoiUpdateDto dto)
+    public void AddPoi(PoiCreateDto dto,CancellationToken token = default)
     {
-        Poi? poi = await _dbContext.Pois.FirstOrDefaultAsync(p => p.DeletedDate == null && p.Id == dto.Id);
+        Poi poi = new Poi(dto.Name, dto.Address, dto.UserId);
+        _dbContext.Pois.AddAsync(poi, token);
+    }
+
+    public async Task UpdatePoiAsync(PoiUpdateDto dto,CancellationToken token = default)
+    {
+        Poi? poi = await _dbContext.Pois.FirstOrDefaultAsync(p => p.Id == dto.Id, cancellationToken: token);
```

Comment:
This is generally discouraged. We should separate this to GetPoi. In that case, update should not be async.